Use gauge instead of counter

scrobbleme opened the pull request:
Pull request for https://github.com/CodeAtCode/Prometheus-Metrics-in-WordPress/issues/5

Please note: This may break existing dashboards as the new value is filtered by gauge and not counter...

Mte90 commented:
Maybe adding a parameter in the url to use a type or another with default counter so we don't break anything?

scrobbleme commented:
Makes sense ...

I'm currently doing, let's say, a major rewrite, as I need some more features... ;) Will push it as soon as possible, so you can see, if it fits into main.

Mte90 commented:
Perfect I will wait :-)

Mte90 commented:
@scrobbleme any updates?

scrobbleme commented:
@Mte90 Yes, I've done a lot of changes, but I'm sure it is not compatible anymore ;)

- refactored the code to use classes for each metric, simplification... 
  - (with some caching)
- added a handy  website health check page 
  - with the ability to generate a Prometheus key

Still not totally satisfied, some clean up, updating documentation, coding...

Maybe have a look at my code (have to check later, if I have committed everything)

https://github.com/scrobbleme/Prometheus-Metrics-in-WordPress/tree/use_classes

Mte90 commented:
It is indeed better then my version.
I am wondering if it is backward compatible without changing the Grafana settings.
As the readme isn't changed and looking at the code I don't see anything that can break something but overall is very cool and better.

Mte90 commented:
So the code seems very good, I think that a bit of backward compatibility will be good (I guess the url parameters is enough) and probably also a new grafana dashboard and it's done to me.

scrobbleme commented:
:+1:

Great, I will see that will have some time end of week or next week to clean up things a litte.

I'm not very familiar with Grafana (and Prometheus). Just able to do some performance metrics now  (with InfluxDB 2) ;)

(Maybe a 2.0 branch is better for the beginning, instead of directly master)

Mte90 commented:
I agree about everything, for the dashboard I will update it if it is needed.

scrobbleme commented:
@Mte90 Can you create a "2.0" branch, so I can create a pull request for this?  

I think it's ready for a "first" version now, but probably still needs some checks/tests...

Mte90 commented:
Branch created and added you as maintainer in the repo :-)
